test/decl/var/variables.swift

```diff
@@ -107,6 +107,7 @@ func test21057425() -> (Int, Int) {
 
 // rdar://problem/21081340
 func test21081340() {
+  func foo() { }
```

Comment:
What is this change testing?

Reply:
The test is verifying that the type of the pattern matches up with the form of the pattern, but the call to an undeclared “foo” is getting in the way.  It wasn’t before because we would prematurely apply coerce the pattern to the given type.